js/index.js

```diff
@@ -41,9 +41,41 @@ window.addEventListener('load', event => {
 
   // Bind the click event of each element to a function
   document.querySelectorAll('.card').forEach(card => {
-    card.addEventListener('click', () => {
-      // TODO: write some code here
-      console.log(`Card clicked: ${card}`);
+    card.addEventListener('click', (event) => {
+      let cardObject = event.target.parentNode;
+      if (cardObject.classList.contains('blocked')) return null;
+
+      memoryGame.pickedCards.length === 0 ? resetCards() : null;
+      memoryGame.pickedCards.push(card.getAttribute("data-card-name"));
+      cardObject.classList.toggle('turned');
+
+      if(memoryGame.pickedCards.length === 2) {
+        if (memoryGame.checkIfPair(memoryGame.pickedCards[0], memoryGame.pickedCards[1])) {
+          let turnedCards = document.querySelectorAll('.turned');
+          turnedCards.forEach((card) => {
+            card.classList.add('blocked');
+          })
+        };
+        updateStats();
+        memoryGame.pickedCards = [];
+      }
+
     });
   });
+ 
 });
+//memoryGame.shuffleCards();
+
+function resetCards() {
+    let turnedCards = document.querySelectorAll(".turned");
+    turnedCards.forEach((card) => {
+      !card.classList.contains('blocked') ? card.classList.remove("turned") : null;
+    })
+}
+
+function updateStats() {
```

Comment:
Great you are storing these blocks of code in functions.
It makes the code more readable.